contracts/VotingRewards.sol

```diff
@@ -0,0 +1,502 @@
+                rewards[i].state != RewardState.Withdrawn
+            ) {
+                rewards[i].state = RewardState.Withdrawn;
+                rewardsVault.transfer(
+                    rewardsToken,
+                    _beneficiary,
+                    rewards[i].amount
+                );
+                collectedRewards = collectedRewards.add(1);
+
+                emit RewardCollected(_beneficiary, rewards[i].amount);
+            }
+        }
+
+        require(collectedRewards >= 1, ERROR_NO_REWARDS);
+    }
+
+    /**
+     * @notice Reward is calculated as the minimum balance between the
+     *         end of an epoch (now) and the balance at the first
+     *         vote in an epoch (in percentage) for each vote happened within the epoch
+     * @param _beneficiary beneficiary
+     * @param _fromBlock block from wich starting looking for votes
+     * @param _toBlock   block to wich stopping looking for votes
+     * @param _missingVotesThreshold number of vote to which is possible to don't vote
+     */
+    function _calculateReward(
+        address _beneficiary,
+        uint64 _fromBlock,
+        uint64 _toBlock,
+        uint256 _missingVotesThreshold
+    ) internal view returns (uint256) {
+        uint256 missingVotes = 0;
+        uint256 minimunBalance = 0;
+        bool forceFirstAssignment = true;
+
+        // voteId starts from 1 in DandelionVoting
+        for (
+            uint256 voteId = dandelionVoting.votesLength() + 1;
+            voteId > 1;
+            voteId--
+        ) {
+            uint256 realVoteId = voteId.sub(1);
+            uint64 startBlock;
+            (, , startBlock, , , , , , , , ) = dandelionVoting.getVote(
+                realVoteId
+            );
+
+            if (startBlock >= _fromBlock && startBlock <= _toBlock) {
+                DandelionVoting.VoterState state = dandelionVoting
+                    .getVoterState(realVoteId, _beneficiary);
+
+                if (state == DandelionVoting.VoterState.Absent) {
+                    missingVotes = missingVotes.add(1);
+                }
+
+                require(
+                    missingVotes <= _missingVotesThreshold,
+                    ERROR_TOO_MANY_MISSING_VOTES
+                );
+
+                uint256 votingTokenBalanceAtVote = MiniMeToken(
+                    dandelionVoting.token()
+                )
+                    .balanceOfAt(_beneficiary, startBlock);
+
+                if (forceFirstAssignment == true) {
+                    forceFirstAssignment = false;
+                    minimunBalance = votingTokenBalanceAtVote;
+                }
+
+                if (votingTokenBalanceAtVote < minimunBalance) {
+                    minimunBalance = votingTokenBalanceAtVote;
+                }
+            }
+
+            // NOTE: avoid "out of epoch" cycles
+            if (startBlock < _fromBlock) break;
+        }
+
+        return (minimunBalance).mul(percentageRewards).div(PCT_BASE);
```

Comment:
Is it worth checking for `minimumBalance` of 0 here first, and just returning that if that's the case, and so only perform the calculation if `minimumBalance` > 0?